pymc3/tests/test_parallel_sampling.py

```diff
@@ -11,9 +11,68 @@
 #   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
+import multiprocessing
+import os
 
+import pytest
 import pymc3.parallel_sampling as ps
 import pymc3 as pm
+import theano
+import theano.tensor as tt
+import numpy as np
+
+
+def test_context():
+    with pm.Model():
+        pm.Normal('x')
+        ctx = multiprocessing.get_context('spawn')
+        pm.sample(tune=2, draws=2, chains=2, cores=2, mp_ctx=ctx)
+
+
+class NoUnpickle:
+    def __getstate__(self):
+        return self.__dict__.copy()
+
+    def __setstate__(self, state):
+        raise AttributeError("This fails")
+
+
+def test_bad_unpickle():
+    with pm.Model() as model:
+        pm.Normal('x')
+
+    with model:
+        step = pm.NUTS()
+        step.no_unpickle = NoUnpickle()
+        with pytest.raises(Exception) as exc_info:
+            pm.sample(tune=2, draws=2, mp_ctx='spawn', step=step,
+                      cores=2, chains=2, compute_convergence_checks=False)
+        assert 'could not be unpickled' in str(exc_info.getrepr(style='short'))
+
+
+@theano.as_op(
+    [
+        tt.dvector if theano.config.floatX == "float64" else tt.fvector,
+        tt.iscalar,
+    ],
+    [tt.dvector if theano.config.floatX == "float64" else tt.fvector],
+)
```

Comment:
Copied from the other PR

```suggestion
tt_vector = tt.TensorType(theano.config.floatX, [False])
@theano.as_op([tt_vector, tt.iscalar], [tt_vector])
```